### PR TITLE
Added module and problem schema for review

### PR DESCRIPTION
**Pull request by dhruv2000**
Please let me know what you guys think in regards to this commit. I will look into JOI validation soon. Also, when would you like me to complete this by? I think we said the next meeting, but is that referring to the one next week with aman, or are we doing that this week?

### CODE REVIEW

**Review by HamishPierpont — COMMENTED** [review]
.

**Review by HamishPierpont — CHANGES_REQUESTED** [review]
Looks solid! A few suggestions:
- `src/models/module.ts`:
  - Remove the comment on line 1
  - Change `Problem` to `problem` on line 13
  - `ObjectId` is actually a [type in MongoDB and mongoose](https://mongoosejs.com/docs/schematypes.html#objectids), so I think you can simplify the problem array to look like this:
  ```javascript
  problem: [mongoose.ObjectId]
  ```
  - Add a newline at the end of the file because GitHub says so
- `src/models/problem.ts`
  - Remove the comment on line 1
  - For `fileExtension`, I don't think `Enumerator` is a [permitted mongoose type](https://mongoosejs.com/docs/schematypes.html). I think we should change the type to `String` and add, next to `type` and `required`, this field:
  ```javascript
  enum: [".java", ".cpp", ".h", ] //I don't know what else could go in here? Maybe ".cxx"?
  ```
  - For `visibility`, I'd replace the min-max validation with `enum: [0, 1, 2]` to avoid any non-integer visibilities from ending up in the DB. This probably would never be an issue, but we might as well have the strictest validation possible.
  - Add a newline at the end of the file because GitHub says so

**Comment by dhruv2000** [comment]
I just made some of those changes. Requesting approval again

**Review by HamishPierpont — APPROVED** [review]
Looks great!

File: src/models/problem.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                type: String,
+                required: true,
+            },
+            // visibility: 0, 1, or 2
+            visibility: {
+                type: Number,
+                required: true,
+                enum: [0,1,2],
+            }
+            /*templatePackage: {
```
Comment (marcgabe15) [inline src/models/problem.ts:63]: Will clarify with professor today

**Review by marcgabe15 — APPROVED** [review]

**Pull request merged**